Tests/SwiftDocCTests/Semantics/MarkupReferenceResolverTests.swift

```diff
@@ -27,4 +27,11 @@ class MarkupReferenceResolverTests: XCTestCase {
         _ = resolver.visit(document)
         XCTAssertEqual(0, resolver.problems.count)
     }
+
+    func testDuplicatedDiagnosticForExtensionFile() throws {
+        let (_, context) = try testBundleAndContext(named: "ExtensionArticleBundle")
+        // Before #733's change, there will be 2 duplicated problems.
+        // And we'll trigger a index out of range crash in DefaultDiagnosticConsoleFormatter
```

Comment:
How about something like this to elaborate on the issue a bit?
```suggestion
        // Before #733, symbols with documentation extension files emitted duplicated problems:
        // - one with a source location in the in-source documentation comment
        // - one with a source location in the documentation extension file.
        // The source range was only valid for one of these diagnostics. This resulted in an index out of range crash in DefaultDiagnosticConsoleFormatter when displaying line that caused the problem to the user
```